cf-check/dump.c

```diff
@@ -154,23 +165,27 @@ static void print_struct_averages(
     else
     {
         // TODO: clean up Averages
+        const char *const *obnames = NULL;
         const Averages *const averages = value.mv_data;
         const time_t last_seen = averages->last_seen;
 
-        JsonElement *all_observables = JsonObjectCreate(observables_max);
-        assert(observables_max <= CF_OBSERVABLES);
+        GetObservableNames(tskey_filename, &obnames);
+        JsonElement *all_observables = JsonObjectCreate(CF_OBSERVABLES);
 
-        for (Observable i = 0; i < observables_max; ++i)
+        for (Observable i = 0; i < CF_OBSERVABLES; ++i)
         {
+            const char *name = obnames[i];
             JsonElement *observable = JsonObjectCreate(4);
             QPoint Q = averages->Q[i];
-            const char *const name = observable_strings[i];
 
-            JsonObjectAppendReal(observable, "q", Q.q);
-            JsonObjectAppendReal(observable, "expect", Q.expect);
-            JsonObjectAppendReal(observable, "var", Q.var);
-            JsonObjectAppendReal(observable, "dq", Q.dq);
-            JsonObjectAppendObject(all_observables, name, observable);
+            if ( name != NULL )
```

Comment:
This check is a left-over, almost certainly we could skip this check. I don't think we actually have a way to know that a measurement has no data other than expectation that all values (expect, q and var) being zero means "no data". (see https://tracker.mender.io/browse/ENT-7461)